### PR TITLE
Add getting tree size and depth

### PR DESCRIPTION
**Pull request by pampi**
Should I expose GetTreeSize() and GetTreeDepth() in KVStore class?

### CODE REVIEW

File: lib/pmem/ARTree.cpp [hunk 1]
```diff
@@ -150,6 +151,47 @@ TreeImpl::TreeImpl(const string &path, const size_t size,
...
+{
+    persistent_ptr<Node256> node256;
+    uint64_t size = 0;
+
+    if (current->type == TYPE256) {
+        // TYPE256
+        node256 = current;
+        for (int i = 0; i < 256; i++) {
+            if (node256->children[i])
+            size += getTreeSize(node256->children[i]);
```
Comment (gjerecze) [inline lib/pmem/ARTree.cpp]: indentation? Could you run clang?

File: lib/pmem/ARTree.cpp [hunk 2]
```diff
@@ -150,6 +151,47 @@ TreeImpl::TreeImpl(const string &path, const size_t size,
...
+// Simple calculation tree size with recursive DFS algorithm
+uint64_t TreeImpl::getTreeSize(persistent_ptr<Node> current)
+{
+    persistent_ptr<Node256> node256;
+    uint64_t size = 0;
+
+    if (current->type == TYPE256) {
+        // TYPE256
+        node256 = current;
+        for (int i = 0; i < 256; i++) {
```
Comment (gjerecze) [inline lib/pmem/ARTree.cpp]: Please use existing ifdef or define a new one for the magic number

File: lib/pmem/ARTree.cpp [hunk 3]
```diff
@@ -150,6 +151,47 @@ TreeImpl::TreeImpl(const string &path, const size_t size,
     }
 }
 
+// Simple calculation tree size with recursive DFS algorithm
+uint64_t TreeImpl::getTreeSize(persistent_ptr<Node> current)
```
Comment (gjerecze) [inline lib/pmem/ARTree.cpp]: I would also propose to add function returning the number of leaves only

File: lib/pmem/ARTree.cpp [hunk 4]
```diff
@@ -150,6 +151,47 @@ TreeImpl::TreeImpl(const string &path, const size_t size,
...
+
+uint8_t TreeImpl::getTreeDepth(persistent_ptr<Node> current)
+{
+    persistent_ptr<Node256> node256;
+    uint8_t depth = 0;
+
+    if (current->type == TYPE256) {
+        // TYPE256
+        node256 = current;
+        for (int i = 0; i < 256; i++) {
```
Comment (gjerecze) [inline lib/pmem/ARTree.cpp]: Same magic number

File: lib/pmem/ARTree.h [hunk 5]
```diff
@@ -145,6 +145,8 @@ class TreeImpl {
     pool<ARTreeRoot> _pm_pool;
     void setClassId(enum ALLOC_CLASS c, unsigned id);
     unsigned getClassId(enum ALLOC_CLASS c);
+    uint64_t getTreeSize(persistent_ptr<Node> current);
```
Comment (gjerecze) [inline lib/pmem/ARTree.h]: Could you also add a public interface in KVStore that would use those?

**Comment by gjerecze** [comment]
> Should I expose GetTreeSize() and GetTreeDepth() in KVStore class?

Please use both (+ leaves only).

File: lib/pmem/ARTree.cpp (hunk 1 above)
Comment (pampi) [reply lib/pmem/ARTree.cpp]: Now it should be fine. (updated with git commit --amend & push --force)

**Pull request merged**